### ISSUE TITLE
"{}" causes `ic()` to break

### ISSUE THREAD

**Issue by notPlancha** [lewinfox/icecream#26]
Because of `glue` (and `cli`), any string that includes the brackets will break, which I feel it's not intentional, since I think I'd want the contents of the variable I'm seeing (similar to print). I first noticed it when working with json.
```R
library(icecream)
ic("{")     # Error 
ic("{p}")  # Error 
var <- "{printthis}"
ic(var)
var2 <- ic(paste("{", "anotherthing")) #Error
jsonexample <- "{'key1': 'value1'}"
ic(jsonexample) #Error
```

I don't think this is intended, and if it is I think there should be a note about this (or an option to not use it).

Depending on how ic.output.function is implemented, the reimplementation could be fixed. Until then, I suggest adding to `print.R`:

```R
output <- paste(prefix, output)
output <- gsub("\\{", "{{", output)
cli::cli_alert_info(output) # TODO: This is ...
```

```R
escape_glue <- \(x) cli::cli_text(gsub("\\{", "{{", x))
escape_glue("{")
escape_glue("{}")
escape_glue("{printthis}")
escape_glue("{'key1': 'value1'}")
escape_glue(paste("{", "anotherthing"))
escape_glue("No brackets")
```

I can create a pull request if you think the suggestion is fine (and enough)

edit: just remembered other types that are not `character` so what I was recommending is not enough

**Comment by lewinfox** [comment]
@notPlancha really appreciate you taking the time to flag this, and to submit the PR. As mentioned in my comment on #27 I don't want to apply a blanket find-and-replace, so I've gone for "try it, if you hit an error escape the curlies". I've also added a couple of unit tests based on the examples you provided.

Would you be interested in doing a bit of testing? If so you can `devtools::install_github("lewinfox/icecream", "bug/curlies")`. Any feedback would be gratefully received.

If I don't hear from you I'll look to do a release in the next week or so.

**Comment by notPlancha** [comment]
looks good to me

**Comment by lewinfox** [comment]
@notPlancha just released v0.2.2 to CRAN with this fix. Any further issues let me know. Thanks again!
